lib/thread/common/export/ecl/thread/common/mutex.hpp

```diff
@@ -10,7 +10,7 @@ namespace common
 class mutex
 {
 public:
-    constexpr mutex();
+    constexpr mutex() { }
```

Comment:
please use `= default` to make emphasis that default construction is what you want. It'll also be inlined.